### PR TITLE
cmdline/parsers: fixed float edge cases and enums

### PR DESCRIPTION
**Pull request by Luyten-Orion**
## Summary
* fixed `parseCli` raising an error on NaN inputs
* fixed `parseCli` for enum values not compiling

## Details
* Added tests for cmdline/parsers.nim
* The float parser test was split into a separate file since  `isNaN` 
doesn't work on the VM.

### CODE REVIEW

File: lib/experimental/cmdline/parsers.nim [hunk 1]
```diff
@@ -80,7 +81,7 @@ proc parseCli*(T: typedesc[SomeFloat], value: string): T =
   ## Implements `parseCli` for all floats.
   let parsed = parseFloat(value)
 
-  if parsed notin low(T)..high(T):
+  if not (parsed in low(T)..high(T) or parsed.isNaN):
```
Comment (alaviss) [inline lib/experimental/cmdline/parsers.nim]:
```suggestion
  if parsed notin low(T)..high(T) and not parsed.isNaN:
```

Use De Morgan's to remove unnecessary nesting.

Though for this case it's better to just remove the range test, since it's always between two `inf`s, making this a no-op.

**Review by zerbina — CHANGES_REQUESTED** [review]
Thanks for the PR!

Since it's a bugfix PR affecting user-facing behaviour, please change the summary such that it focuses on the *what* (the *why* and *how* go into the details section). For example:
```
* fixed `parseCli` for float values reporting an error for `nan` inputs
* fixed `parseCli` for enum values not compiling
```

The idea is that someone interested in whether the changes/fixes affect them can quickly spot this by only looking at the summary.

**Review by zerbina — APPROVED** [review]

File: lib/experimental/cmdline/parsers.nim [hunk 2]
```diff
@@ -78,13 +78,7 @@ proc parseCli*(T: typedesc[SomeInteger], value: string): T =
...
 proc parseCli*(T: typedesc[SomeFloat], value: string): T =
   ## Implements `parseCli` for all floats.
-  let parsed = parseFloat(value)
-
-  if parsed notin low(T)..high(T):
-    raise newException(ValueError):
-      $parsed & " is not in the range of " & $low(T) & ".." & $high(T)
-
-  result = T(parsed)
+  parseFloat(value)
```
Comment (alaviss) [inline lib/experimental/cmdline/parsers.nim]: Needs conversion to target type or this will fail for float32.

**Review by alaviss — CHANGES_REQUESTED** [review]

File: lib/experimental/cmdline/parsers.nim (hunk 2 above)
Comment (zerbina) [reply lib/experimental/cmdline/parsers.nim]: It presently won't (`float` is implicitly-convertible to `float32`), but having an explicit conversion to be future proof is better.

**Review by alaviss — APPROVED** [review]

**Review by saem — APPROVED** [review]
